index.js

```diff
@@ -100,6 +100,21 @@ class PQueue {
 		return this._pendingCount < this._concurrency;
 	}
 
+	get concurrency() {
+		return this._concurrency;
+	}
+
+	set concurrency(newConcurrency) {
+		if (!(typeof newConcurrency === 'number' && newConcurrency >= 1)) {
+			throw new TypeError(`Expected \`concurrency\` to be a number from 1 and up, got \`${newConcurrency}\` (${typeof newConcurrency})`);
+		}
+		const prevConcurrency = this._concurrency;
+		this._concurrency = newConcurrency;
+		if (newConcurrency > prevConcurrency) {
+			while (this._tryToStartAnother()) {} // eslint-disable-line no-empty
```

Comment:
If the queue is empty and/or paused, won't you get caught in a tight loop here? Could you add test(s) for this scenario to ensure that continued processing is possible. My concern is you'd never be able to un-pause or add items once you reach this condition since the caller will be waiting on the setter to complete. Similarly, long running queue items could result in a long execution lag.

Returning a promise may be ideal with checking done in an interval. This would allow execution to continue, with consumer to wait / respond if they desire.

Reply:
Should be cool since js function calls are atomic ✌️

Reply:
> My concern is you'd never be able to un-pause or add items once you reach this condition since the caller will be waiting on the setter to complete.

Nah. If the queue is empty/paused it returns `false`.

> Similarly, long running queue items could result in a long execution lag.

The `run` function is a Promise, it won't block further executions.